spec/javascripts/health-care/components/form-elements/GrowableTable.spec.jsx

```diff
@@ -0,0 +1,174 @@
+      SkinDeep.shallowRender(
+        <GrowableTable rows component={func} createRow={func} onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `rows` was not specified in `GrowableTable`/);
+    });
+  });
+
+  describe('row management', () => {
+    let growableTable;
+    let currentId = 0;
+    const createRow = () => { return { value: currentId++ }; };
+    const rows = [
+      { key: 'k1', value: 'v1' },
+      { key: 'k2', value: 'v2' }
+    ];
+    const onRowsUpdate = sinon.spy();
+
+    before(() => {
+      // It's perfectly fine in this test to reuse the rendered component. Do that
+      // cause it cuts the test time from 1s down to ~0.1s.
+      growableTable = ReactTestUtils.renderIntoDocument(
+        <GrowableTable
+            component={Row}
+            createRow={createRow}
+            rows={rows}
+            onRowsUpdate={onRowsUpdate}/>
+      );
+    });
+
+    afterEach(() => {
+      onRowsUpdate.reset();
+    });
+
+    it('Add Button adds new row w/o distubing existing entries', () => {
+      const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(growableTable, 'button');
+      const addButton = buttons.find((element) => { return element.textContent === 'Add'; });
+      expect(addButton).to.not.be.undefined;
+      ReactTestUtils.Simulate.click(addButton);
+      sinon.assert.calledOnce(onRowsUpdate);
+
+      // Ensure original rows are untouched.
+      const update = onRowsUpdate.firstCall.args[0];
+      expect(update).to.have.length(rows.length + 1);
+      expect(rows).to.eql(update.slice(0, rows.length));
+
+      // Ensure new item has the key element and uses the createRow function.
+      const newItem = update[rows.length];
+      expect(newItem).to.have.property('key');
+      expect(newItem).to.have.property('value').and.equal(currentId - 1);
+
+      // Ensure second row update has different key.
+      ReactTestUtils.Simulate.click(addButton);
+      sinon.assert.calledTwice(onRowsUpdate);
+      const newItem2 = onRowsUpdate.secondCall.args[0][rows.length];
+      expect(newItem2).to.have.property('value').and.equal(currentId - 1).and.not.eql(newItem.value);
+      expect(newItem2).to.have.property('key').and.to.not.eql(newItem.key);
+    });
+
+    it('Remove Button removes its associated row', () => {
+      const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(growableTable, 'button');
+      const removeButtons = buttons.filter((element) => { return element.textContent === 'Remove'; });
+      expect(removeButtons).to.have.length(2);
+
+      // Trigger the first remove should leave only the second element.
+      ReactTestUtils.Simulate.click(removeButtons[0]);
+      sinon.assert.calledOnce(onRowsUpdate);
+      let update = onRowsUpdate.firstCall.args[0];
+      expect(update).to.have.length(1);
+      expect(update).to.eql(rows.slice(1, rows.length));
+
+      // Trigger the second remove should leave only the second element.
```

Comment:
He he he...good question. The handler is just a mock using spy. Thus, the component fires off a status update which does...absolutely nothing. Therefor it never sends a prop update down and the component never changes state.